src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java

```diff
@@ -175,17 +168,12 @@ void testBasicDelivery() throws ExecutionException, InterruptedException {
             sendFuture.get();
         }
 
-        TestUtils.waitForCondition(
-            () -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size(),
-            15000,
-            "All requests received by HTTP server"
-        );
-        log.info("Recorded request bodies: {}", bodyRecorderHandler.recorderBodies());
-        assertIterableEquals(expectedBodies, bodyRecorderHandler.recorderBodies());
+        await("All expected requests received by HTTP server")
+                .atMost(Duration.ofSeconds(15)).pollInterval(Duration.ofMillis(100))
+                .until(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size());
 
-        log.info("{} HTTP requests were expected, {} were successfully delivered",
-            expectedBodies.size(),
-            bodyRecorderHandler.recorderBodies().size());
+        assertThat(bodyRecorderHandler.recorderBodies())
+                .containsExactlyInAnyOrderElementsOf(expectedBodies);
```

Comment:
here too the previous test seemed to care about order

Reply:
Fixed